### PR TITLE
(IMAGES-465) Create Windows-2012r2 Japanese Packerized Template

### PR DESCRIPTION
**Pull request by jcoconnor**
As it says in the title.
Some additional maintenance was also needed to fixup the oracle Certs were updated due to an interesting wrinkle around the Virtual Box 5.1.12 release where the driver cert had expired.
All build machines have had to be updated to Virtual Box 5.1.16 to support this build.

Some additional issues have also been addressed in this PR (IMAGES-501 & IMAGES-502)

### CODE REVIEW

**Comment by jcoconnor** [comment]
This appears to build correctly and lands on ```int-resources```
Will do some checking to see if its consistent.
Also need to get the Amazon S3 bucket build sorted ([IMAGES-483](https://tickets.puppetlabs.com/browse/IMAGES-483))

**Comment by jcoconnor** [comment]
Adding two additional fixes, so hold off merge please.

**Comment by jcoconnor** [comment]
This image is building and up-loading to s3://puppet-vagrant-windows.
I have successfully downloaded it configured it in vagrant and connected to it (also using ssh).

So this PR is ready for review/merge.

Ping @puppetlabs/agent-team @glennsarti @jpogran

**Comment by jcoconnor** [comment]
Please don't merge - I'm going to update this and the ```vmpooler``` image to WMF5 for Japanese only. [IMAGES-509](https://tickets.puppetlabs.com/browse/IMAGES-509)

File: scripts/windows/windows-env.ps1 [hunk 1]
```diff
@@ -174,3 +174,23 @@ Function Install_Win_Patch
   Start-Process -Wait "wusa.exe" -ArgumentList "$ENV:TEMP\$PatchFilename /quiet /norestart"
   Write-Host "Patch Installed"
 }
+
+# Helper function to delete file, with try/catch to ignore errors.
+# This function is used in both the clean host and clean-disk scripts.
+
+Function Delete_File_Forcefully
```
Comment (Iristyle) [inline scripts/windows/windows-env.ps1]: Nit - PowerShell functions should use dashes, not underscores - and should follow `Verb-Noun` style naming.

File: scripts/windows/windows-env.ps1 [hunk 2]
```diff
@@ -174,3 +174,23 @@ Function Install_Win_Patch
...
+{
+  $filetodelete = $args[0]
+
+  try {
+    if(Test-Path $filetodelete) {
+        Write-Host "Removing $filetodelete"
+        Takeown /d Y /R /f $filetodelete
+        Icacls $filetodelete /GRANT:r administrators:F /T /c /q  2>&1 | Out-Null
```
Comment (Iristyle) [inline scripts/windows/windows-env.ps1]: This seems pretty heavy-handed - any idea on why this is necessary?

Comment (jcoconnor) [inline scripts/windows/windows-env.ps1]: Yes - was picked up from the original examples by Matt Wrock some time ago.
Will go back and check the article, but I think it was belt and braces approach to clean out the temporary areas.

File: scripts/windows/windows-env.ps1 (hunk 1 above)
Comment (jcoconnor) [inline scripts/windows/windows-env.ps1]: OK

**Comment by jcoconnor** [comment]
@pmcmaw Has run ```stdlib``` acceptance tests using the new Vagrant image.
The only issue encountered was a known issue with ```puppet agent -t``` that is not related to the image.

File: scripts/windows/windows-env.ps1 (hunk 2 above)
Comment (jcoconnor) [inline scripts/windows/windows-env.ps1]: @Iristyle The reference for this code is at: http://www.hurryupandwait.io/blog/creating-a-windows-server-2016-vagrant-box-with-chef-and-packer under "Cleaning Up".

I agree its heavy duty, but have found it to be a lot more reliable at cleaning files in temp, especially with the ```try/catch``` to trap and ignore errors. Was one of the "random" failures that would hit the build occasionally.

I also lifted the disk zeroing code as it was much faster than ```sdelete``` sysinternals utility.

```Write-Host "Cleaning Temp Files"
try {
  Takeown /d Y /R /f "C:\Windows\Temp\*"
  Icacls "C:\Windows\Temp\*" /GRANT:r administrators:F /T /c /q  2>&1
  Remove-Item "C:\Windows\Temp\*" -Recurse -Force -ErrorAction SilentlyContinue
} catch { }

Write-Host "Optimizing Drive"
Optimize-Volume -DriveLetter C

Write-Host "Wiping empty space on disk..."
$FilePath="c:\zero.tmp"
$Volume = Get-WmiObject win32_logicaldisk -filter "DeviceID='C:'"
$ArraySize= 64kb
$SpaceToLeave= $Volume.Size * 0.05
$FileSize= $Volume.FreeSpace - $SpacetoLeave
$ZeroArray= new-object byte[]($ArraySize)
 
$Stream= [io.File]::OpenWrite($FilePath)
try {
   $CurFileSize = 0
    while($CurFileSize -lt $FileSize) {
        $Stream.Write($ZeroArray,0, $ZeroArray.Length)
        $CurFileSize +=$ZeroArray.Length
    }
}
finally {
    if($Stream) {
        $Stream.Close()
    }
}
 
Del $FilePath
```

**Comment by jcoconnor** [comment]
I have removed the WMF5 install [IMAGES-509](https://tickets.puppetlabs.com/browse/IMAGES-509) for a separate PR as there are a few other issues to be resolved before I add it.

So this PR is effectively as it was yesterday with a minor fix requested by @Iristyle 

As noted by above, it has also been used by @pmcmaw for modules/i18n testing - see [FM-6099](https://tickets.puppetlabs.com/browse/FM-6099)

**Comment by McdonaldSeanp** [comment]
👍

**Pull request merged**